BJS2-1516 finished task (Following Count). BJS2-1510 finished task (Followers Count).

starkoracia opened the pull request:
```
feature(SubscriptionController, SubscriptionServiceImpl, SubscriptionService) {
  + getFollowersCount() method
}
feature(SubscriptionServiceImpl) {
  + validateUserIdIsPositive(long userId) added
}
refactor(SubscriptionServiceImpl) {
  -+ validateUserIds()
}
refactor(SubscriptionRepository) {
  -+ made JpaRepository instead of Crud
}
feature(SubscriptionController, SubscriptionService, SubscriptionServiceImpl) {
  + getFollowersCount() method added
}
-----
test(SubscriptionServiceImplTest, SubscriptionControllerTest) {
  + relevant tests added
}
```

starkoracia committed:
BJS2-1414 test refactor.

refactor() {
 -+ Move test's fields to constants
}

starkoracia committed:
BJS2-1414 test refactor.

refactor() {
 -+ Move test's fields to constants
}

starkoracia committed:
BJS2-1516 finished task (Followees Count).

feature(SubscriptionController, SubscriptionServiceImpl, SubscriptionService) {
 + getFollowersCount() method
}
feature(SubscriptionServiceImpl) {
 + validateUserIdIsPositive(long userId) added
}
refactor(SubscriptionServiceImpl) {
 -+ validateUserIds()
}
refactor(SubscriptionRepository) {
 -+ Made JpaRepository instead of Crud
}
-----
test(SubscriptionServiceImplTest, SubscriptionControllerTest) {
 + Relevant tests added
}

starkoracia committed:
Merge branch 'basilisk-master-bc2' into basilisk-master-bc2-BJS2-1414

starkoracia committed:
Merge branch 'basilisk-master-bc2-BJS2-1414-feature-BJS2-1516' into basilisk-master-bc2-BJS2-1414

# Conflicts:
#	src/test/java/school/faang/user_service/controller/subscription/SubscriptionControllerTest.java

starkoracia committed:
Merge branch 'basilisk-master-bc2' into basilisk-master-bc2-BJS2-1414

starkoracia committed:
BJS2-1414 refactor.

refactor(SubscriptionController, SubscriptionControllerTest, SubscriptionService, SubscriptionControllerTest, SubscriptionServiceImplTest.java) {
 -+ Changed name of getFollowersCount to getFollowingCount
}

starkoracia committed:
BJS2-1510 finished task (Followers Count).

feature(SubscriptionController, SubscriptionService, SubscriptionServiceImpl) {
  + getFollowersCount() method added
}
test() {
  + relevant tests added
}

PavelAP96 commented on src/main/java/school/faang/user_service/service/subscription/SubscriptionServiceImpl.java:
```diff
@@ -27,19 +27,36 @@ public void unfollowUser(long followerId, long followeeId) {
         }
     }
 
+    private void validateUserIdIsPositive(long userId) {
+        if (userId <= 0) {
+            throw new DataValidationException("User identifiers must be positive numbers");
+        }
+    }
+
```
я бы просто назвал validateUserId, потому что если добавятся еще какие - то проверки на id, то придется менять название

starkoracia replied:
++

PavelAP96 commented on src/test/java/school/faang/user_service/controller/subscription/SubscriptionControllerTest.java:
```diff
@@ -9,45 +9,56 @@
 @ExtendWith(MockitoExtension.class)
 class SubscriptionControllerTest {
 
+    private static final long VALID_FOLLOWER_ID = 5L;
+    private static final long VALID_FOLLOWEE_ID = 2L;
```
я бы лучше объявил в блоке `@BeforeEach`, а не делал константами

starkoracia replied:
++

PavelAP96 commented on src/test/java/school/faang/user_service/controller/subscription/SubscriptionControllerTest.java:
```diff
@@ -9,45 +9,56 @@
+
+        verify(subscriptionService).getFollowingCount(VALID_FOLLOWER_ID);
     }
 
```
getFollowersCount не покрыт тестами

starkoracia replied:
++

PavelAP96 commented on src/test/java/school/faang/user_service/service/subscription/SubscriptionServiceImplTest.java:
```diff
@@ -8,13 +8,20 @@
 @ExtendWith(MockitoExtension.class)
 class SubscriptionServiceImplTest {
 
+    private static final long VALID_FOLLOWER_ID = 5L;
+    private static final long VALID_FOLLOWEE_ID = 2L;
+    private static final long INVALID_FOLLOWER_ID = -1L;
+    private static final long INVALID_FOLLOWEE_ID = -2L;
+    private static final long SAME_USER_ID = 1L;
```
я бы лучше объявил в блоке @BeforeEach, а не делал константами
более распостраненный метод

starkoracia replied:
Просто все это перенести в блок @BeforeEach?

starkoracia replied:
Аха) У тебя получилась ссылка на какого то чувака в github )) `@BeforeEach`

starkoracia replied:
И у меня))

starkoracia replied:
++

starkoracia committed:
BJS2-1414 refactor

refactor(SubscriptionServiceImpl) {
  -+ renamed validateUserIdIsPositive() method to validateUserId()
}

refactor(SubscriptionControllerTest, SubscriptionServiceImplTest) {
  -+ changed test constant variables to non-constant and move to @BeforeEach init() method.
}

refactor(SubscriptionServiceImplTest) {
  -+ added shouldReturnFollowersCountWhenFolloweeIdIsValid() method to cover subscriptionController.getFollowersCount() method.
}

starkoracia committed:
Merge branch 'basilisk-master-bc2' into basilisk-master-bc2-BJS2-1414
